### PR TITLE
feat(profiling): fetch flamegraph data on route

### PR DESCRIPTION
**Pull request by JonasBa**
Fetches the flamegraph and initializes the state

### CODE REVIEW

File: static/app/views/profiling/flamegraph.tsx [hunk 1]
```diff
@@ -1,14 +1,83 @@
...
+
+function fetchFlamegraphs(
+  api: Client,
+  eventId: string,
+  projectId: Project['id'],
+  organization: Organization
+): Promise<ProfileGroup> {
+  return api
+    .requestPromise(
+      `/organizations/${organization.slug}/${projectId}/profiling/profiles/${eventId}`,
```
Comment (Zylphrex) [inline static/app/views/profiling/flamegraph.tsx:29]: This should be the correct endpoint. Note the trailing `/` is required.

```suggestion
      `/projects/${organization.slug}/${projectId}/profiling/profiles/${eventId}/`,
```

File: static/app/views/profiling/flamegraph.tsx [hunk 2]
```diff
@@ -1,14 +1,83 @@
...
       <FlamegraphPreferencesProvider>
-        <FullScreenFlamegraphContainer>{null}</FullScreenFlamegraphContainer>
+        <FlamegraphThemeProvider>
+          <FullScreenFlamegraphContainer>
+            {requestState === 'errored' ? (
+              <Alert type="error" icon={<IconFlag size="md" />}>
+                {t('Unable to load profiles')}
+              </Alert>
+            ) : requestState === 'loading' ? (
+              <TransparentLoadingMask visible />
```
Comment (Zylphrex) [inline static/app/views/profiling/flamegraph.tsx:73]: I think you'll want to use the `<LoadingIndicator />` here. `<TransparentLoadingMask />` is typically used only for charts.

File: static/app/components/profiling/flamegraph.tsx [hunk 3]
```diff
@@ -26,7 +26,7 @@ function Flamegraph(props: FlamegraphProps): React.ReactElement {
   const canvasPoolManager = useMemo(() => new CanvasPoolManager(), []);
 
   const [flamegraph, setFlamegraph] = useState(
-    new FlamegraphModel(props.profiles.profiles[0], 0, {
+    new FlamegraphModel(props.profiles.profiles[props.profiles.activeProfileIndex], 0, {
       inverted: view === 'bottom up',
       leftHeavy: sorting === 'left heavy',
     })
```
Comment (Zylphrex) [inline static/app/components/profiling/flamegraph.tsx:32]: Is the second argument to `FlamegraphModel` used anywhere?

```suggestion
    new FlamegraphModel(
      props.profiles.profiles[props.profiles.activeProfileIndex],
      props.profiles.activeProfileIndex,
      {
        inverted: view === 'bottom up',
        leftHeavy: sorting === 'left heavy',
      }
    )
```

Also, I just noticed but why is there an `useEffect` further down always uses the 0th index?

**Review by Zylphrex — APPROVED** [review]

File: static/app/components/profiling/flamegraph.tsx (hunk 3 above)
Comment (JonasBa) [reply static/app/components/profiling/flamegraph.tsx]: Yes, it is! I fixed it :)

**Pull request merged**